src/rdf.rs

```diff
@@ -0,0 +1,275 @@
+
+use std::collections::HashMap;
+
+use crate::vocab::{dcat, dcat_mqa, dcterms, dqv};
+
+// Parse Turtle RDF and load into store
+pub fn parse_turtle(turtle: String) -> Result<Store, StorageError> {
+    info!("Loading turtle graph");
+
+    let store = Store::new()?;
+    match store.load_graph(
+        turtle.as_ref(),
+        GraphFormat::Turtle,
+        GraphNameRef::DefaultGraph,
+        None,
+    ) {
+        Ok(_) => info!("Graph loaded successfully"),
+        Err(e) => error!("Loading graph failed {}", e),
+    }
+
+    Ok(store)
+}
+
+// Retrieve datasets
+pub fn list_datasets(store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        None,
+        Some(rdf::TYPE),
+        Some(dcat::DATASET_CLASS.into()),
+        None,
+    )
+}
+
+// Retrieve distributions of a dataset
+pub fn list_distributions(dataset: NamedNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((dataset).into()),
+        Some((dcat::DISTRIBUTION).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve access urls of a distribution
+pub fn list_access_urls(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcat::ACCESS_URL).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve download urls of a distribution
+pub fn list_download_urls(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcat::DOWNLOAD_URL).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve distribution formats
+pub fn list_formats(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcterms::FORMAT).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve dataset namednode
+pub fn get_dataset_node(store: &Store) -> Option<NamedNode> {
+    list_datasets(&store).next().and_then(|d| match d {
+        Ok(Quad {
+            subject: Subject::NamedNode(n),
+            ..
+        }) => Some(n),
+        _ => None,
+    })
+}
+
+// Map GEO location method
+fn map_format_to_head(format_uri: String) -> String {
+    let fmt = format_uri.split("/").last().unwrap();
+    match fmt {
+        "WMS_SRVC" => "WMS".to_string(),
+        "WFS_SRVC" => "WFS".to_string(),
+        "WCS_SRVC" => "WCS".to_string(),
+        _ => "HEAD".to_string(),
+    }
+}
+
+// Extract accessURLs and downloadURLs from dataset
+pub fn extract_urls_from_distribution(
+    dist_node: NamedOrBlankNodeRef,
+    store: &Store,
+) -> Vec<HashMap<String, String>> {
+    let mut urls = Vec::new();
+
+    let mut url_extract = HashMap::new();
+    url_extract.insert("distribution_uri".to_string(), dist_node.to_string());
+
+    // Map format to HEAD
+    let head = list_formats(dist_node, store).next().map_or_else(
+        || "HEAD".to_string(),
```

Comment:
Unødvendig med en funksjon for default her. Se https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or
```suggestion
    let head = list_formats(dist_node, store).next().map_or(
        "HEAD".to_string(),
```

Reply:
Arguments passed to map_or are eagerly evaluated; if you are passing the result of a function call, it is recommended to use [map_or_else](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_or_else), which is lazily evaluated.

Reply:
Om default value er resultatet av en funksjon, er det unødvendig å kalle funksjonen og gjøre masse arbeid dersom `Option` er `Some(_)` og default verdien ikke blir brukt. Mer her er det bare en string, så ikke mye unødvendig som blir gjort (bare `to_string()`) dersom default ikke blir brukt.